openedx/features/course_experience/api/v1/views.py

```diff
@@ -106,6 +77,52 @@ def reset_course_deadlines(request):
         raise UnableToResetDeadlines from reset_deadlines_exception
 
 
+@api_view(["POST"])
+@authentication_classes(
+    (
+        JwtAuthentication,
+        BearerAuthenticationAllowInactiveUser,
+        SessionAuthenticationAllowInactiveUser,
+    )
+)
+@permission_classes((IsAuthenticated,))
+def reset_all_relative_course_deadlines(request):
```

Comment:
The function names are changing unnecessarily across the three cases.

The original was/is:  

`def reset_course_deadlines(request)`

The refactored helper is:

`def reset_deadlines_for_course(request, course_key, research_event_data={})`
The new build version is:

`def reset_all_relative_course_deadlines(request)`

I think we should have something like:

`def reset_course_deadlines(request)`
`def reset_course_deadlines(request, course_key, research_event_data={})`
`def reset_all_course_deadlines(request)`

I think it would probably be useful to have the core function of resetting in bulk be:

`def reset_bulk_course_deadlines(coursekey_list)`

This would make it available without passing in a request.

Reply:
Yes, you are right.  I think that that naming convention you suggest makes sense.  Ideally, I think, we wouldn't pass the request into the utility method because it couples it to the request cycle.  However, this code is already messy because we're passing around the research event which is definitely non-standard, probably an anti-pattern.